Upgrade pnpm so it won't error in nodejs 20

### System Info

-

### Details

pnpm/pnpm#6424

### Reproduce link

_No response_

### Reproduce Steps

-

I pushed some changes to branch `try-node-20` https://github.com/web-infra-dev/rspack/tree/try-node-20. I haven't got a clue what pnpm 8 is doing, any help is appreciated 😅 

It's currently failing on `pnpm run test:example`:

```
examples/loader-compat build$ rspack build
│ error[internal]: Napi Error: GenericFailure - Thread Loader (Worker 0)
│ Cannot find module 'ajv/dist/compile/codegen'
│ Require stack:
│ - /Users/bytedance/github/rspack/node_modules/babel-loader/node_modules/ajv-formats/dist/limit.js
│ - /Users/bytedance/github/rspack/node_modules/babel-loader/node_modules/ajv-formats/dist/index.js
│ - /Users/bytedance/github/rspack/node_modules/babel-loader/node_modules/schema-utils/dist/validate.js
```

We have ajv 6 somewhere but it requires ajv 7+

--- a/.github/workflows/reusable-build.yml
+++ b/.github/workflows/reusable-build.yml
@@ -213,7 +213,7 @@ jobs:
     strategy:
       fail-fast: false # Build and test everything so we can look at all the errors
       matrix:
-        node: [16, 18]
+        node: [16, 18, 20]
     name: Test Node ${{ matrix.node }}
     env:
       PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: true

--- a/.npmrc
+++ b/.npmrc
@@ -1,3 +1,9 @@
 node-linker=hoisted
 registry=https://registry.npmjs.org/
 strict-peer-dependencies=false
+
+# Change pnpm 8 settings to pnpm 7
+auto-install-peers=false
+dedupe-peer-dependents=false
+resolve-peers-from-workspace-root=false
+resolution-mode='highest'

--- a/crates/rspack_core/Cargo.toml
+++ b/crates/rspack_core/Cargo.toml
@@ -23,7 +23,7 @@ itertools = { workspace = true }
 mime_guess = { workspace = true }
 nodejs-resolver = { version = "0.1.0" }
 once_cell = { workspace = true }
-oxc_resolver = { version = "0.5.3", features = ["tracing-subscriber"] }
+oxc_resolver = { version = "0.5.3" }
 paste = { workspace = true }
 petgraph = { version = "0.6.3", features = ["serde-1"] }
 rayon = { workspace = true }

--- a/package.json
+++ b/package.json
@@ -36,7 +36,7 @@
     "url": "https://github.com/web-infra-dev/rspack"
   },
   "engines": {
-    "pnpm": "7.32.0"
+    "pnpm": "8.10.5"
   },
   "devDependencies": {
     "@rspack/cli": "workspace:*",
@@ -71,5 +71,5 @@
     "sass-embedded-win32-ia32": "1.58.3",
     "sass-embedded-win32-x64": "1.58.3"
   },
-  "packageManager": "pnpm@7.32.0"
+  "packageManager": "pnpm@v8.10.5"
 }

